superset-frontend/src/SqlLab/components/RunQueryActionButton/index.tsx

```diff
@@ -146,7 +146,9 @@ const RunQueryActionButton = ({
               ),
               trigger: 'click',
             }
-          : { buttonStyle: 'primary' })}
+          : {
+              buttonStyle: shouldShowStopBtn ? 'danger' : 'primary',
```

Comment:
I talked with @kasiazjc about this (she's out today), and she said we should make sure it's in line with the button in Explore. Looking at the code it appears we're using `warning` instead of `danger` there:
https://github.com/apache/superset/blob/3c381c5361c16732ebf98b74c95c0ad1516a4a39/superset-frontend/src/explore/components/RunQueryButton/index.tsx#L44

This renders the following button when Explore is fetching data:

![image](https://user-images.githubusercontent.com/33317356/235653441-df93b019-e929-4a3a-a958-bf59bbeaf9f4.png)

So if we change this to `warning` and also change the icon above from `fa fa-stop` to `fa fa-stop-circle-o` then we get a very similar look and feel:

![image](https://user-images.githubusercontent.com/33317356/235652789-5fac5e55-a221-47dc-a7a8-a284e6f63736.png)

For reference, here's the button with the original icon class and `danger` style:
![image](https://user-images.githubusercontent.com/33317356/235654042-b3006400-813c-4d85-b89f-53ef274852fd.png)

Reply:
@villebro Made the changes as requested, please take a look.

Reply:
Thanks @curious86 ! After chatting with @kasiazjc , we arrived at the following minor adjustment:
- Let's keep the `fa fa-stop` class in the SQL Lab button icon.
- Let's swap out `fa fa-stop-circle-o` in the Explore button for `fa fa-stop`

After that good to go!